src/ir_def/check_ir.ml

```diff
@@ -1105,6 +1108,14 @@ and gather_dec env scope dec : scope =
     let val_info = {typ = T.Mut t; const = false; loc_known = env.lvl = TopLvl} in
     let ve = T.Env.add id val_info scope.val_env in
     { val_env = ve }
+  | RefD (id, t, lexp) ->
```

Comment:
It might be nicer if this was similar to VarD and used `t` to describe the contents of the mutable cell, not the type of the cell. Then you could also share the code of the two branches quite nicely for gather_dec